src/viewers/VivViewer.js

```diff
@@ -78,13 +89,55 @@ export default class VivViewer extends PureComponent {
     return viewState;
   }
 
+  componentDidUpdate(prevProps) {
+    const { props } = this;
+    const { views } = props;
+    // Only update state if the previous viewState prop does not match the current one
+    // so that people can update viewState
+    const changed = views
+      .filter(view => {
+        const prevViewState = prevProps.viewStates.find(
+          viewState => viewState.id === view.id
+        );
+        const currViewState = props.viewStates.find(
+          viewState => viewState.id === view.id
+        );
+        return (
+          currViewState && !areViewStatesEqual(currViewState, prevViewState)
+        );
+      })
+      .map(view => view.id);
+    if (changed.length) {
+      const { viewStates: prevStates } = this.state;
+      const viewStates = {};
+      views.forEach(view => {
+        if (changed.includes(view.id)) {
+          const { height, width } = view;
+          const viewState = props.viewStates.find(v => v.id === view.id);
+          viewStates[view.id] = view.filterViewState({
+            viewState: {
+              ...viewState,
```

Comment:
With the way it's currently implemented, if a view doesn't have a view state in the props, the view state in `this.state` remains untouched. This suggestion would put a view state without `zoom` or `target` into `this.state`. I think that the currently implemented outcome is better. It is unlikely that this situation will arise because the current viewers all set a default view state if nothing is specified.